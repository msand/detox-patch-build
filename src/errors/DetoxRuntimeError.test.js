const _ = require('lodash');
const DetoxRuntimeError = require('./DetoxRuntimeError');

describe(DetoxRuntimeError, () => {
  it('should format all fields to a single message', () => {
    _.forEach(varietiesOfInstantiation(), (error, description) => {
      expect(error.toString()).toMatchSnapshot(`.toString() - ${description}`);
      expect(() => { throw error; }).toThrowErrorMatchingSnapshot(`throw error - ${description}`);
    });
  });

  function varietiesOfInstantiation() {
    return {
      'no args': new DetoxRuntimeError(),
      'empty object': new DetoxRuntimeError({}),
      'only message': new DetoxRuntimeError({
        message: `The video is not being recorded on device (${'emulator-5554'}) at path: ${'/sdcard/712398.mp4'}`,
      }),
      'message with hint': new DetoxRuntimeError({
        message: 'Detox adapter to Jest is malfunctioning.',
        hint: `Make sure you register it as Jasmine reporter inside init.js:\n` +
              `-------------------------------------------------------------\n` +
              'jasmine.getEnv().addReporter(adapter);',
      }),
      'message with debug info': new DetoxRuntimeError({
        message: 'no filename was given to constructSafeFilename()',
        debugInfo: 'the arguments were: ' + JSON.stringify({
          prefix: 'detox - ',
          trimmable: undefined,
          suffix: undefined,
        }, null, 2),
      }),
      'message with hint and debug info': new DetoxRuntimeError({
        message: `Invalid test summary was passed to detox.beforeEach(testSummary)` +
        '\nExpected to get an object of type: { title: string; fullName: string; status: "running" | "passed" | "failed"; }',
        hint: 'Maybe you are still using an old undocumented signature detox.beforeEach(string, string, string) in init.js ?' +
        '\nSee the article for the guidance: ' +
        'https://github.com/wix/detox/blob/master/docs/APIRef.TestLifecycle.md',
        debugInfo: `testSummary was: ${JSON.stringify('test name')}`,
      }),
    };
  }
});
