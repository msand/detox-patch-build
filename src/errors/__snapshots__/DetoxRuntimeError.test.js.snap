// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetoxRuntimeError should format all fields to a single message: .toString() - empty object 1`] = `
"
DetoxRuntimeError: "
`;

exports[`DetoxRuntimeError should format all fields to a single message: .toString() - message with debug info 1`] = `
"
DetoxRuntimeError: no filename was given to constructSafeFilename()

the arguments were: {
  \\"prefix\\": \\"detox - \\"
}"
`;

exports[`DetoxRuntimeError should format all fields to a single message: .toString() - message with hint 1`] = `
"
DetoxRuntimeError: Detox adapter to Jest is malfunctioning.

HINT: Make sure you register it as Jasmine reporter inside init.js:
-------------------------------------------------------------
jasmine.getEnv().addReporter(adapter);"
`;

exports[`DetoxRuntimeError should format all fields to a single message: .toString() - message with hint and debug info 1`] = `
"
DetoxRuntimeError: Invalid test summary was passed to detox.beforeEach(testSummary)
Expected to get an object of type: { title: string; fullName: string; status: \\"running\\" | \\"passed\\" | \\"failed\\"; }

HINT: Maybe you are still using an old undocumented signature detox.beforeEach(string, string, string) in init.js ?
See the article for the guidance: https://github.com/wix/detox/blob/master/docs/APIRef.TestLifecycle.md

testSummary was: \\"test name\\""
`;

exports[`DetoxRuntimeError should format all fields to a single message: .toString() - no args 1`] = `
"
DetoxRuntimeError: "
`;

exports[`DetoxRuntimeError should format all fields to a single message: .toString() - only message 1`] = `
"
DetoxRuntimeError: The video is not being recorded on device (emulator-5554) at path: /sdcard/712398.mp4"
`;

exports[`DetoxRuntimeError should format all fields to a single message: throw error - empty object 1`] = `"DetoxRuntimeError: "`;

exports[`DetoxRuntimeError should format all fields to a single message: throw error - message with debug info 1`] = `
"DetoxRuntimeError: no filename was given to constructSafeFilename()

the arguments were: {
  \\"prefix\\": \\"detox - \\"
}"
`;

exports[`DetoxRuntimeError should format all fields to a single message: throw error - message with hint 1`] = `
"DetoxRuntimeError: Detox adapter to Jest is malfunctioning.

HINT: Make sure you register it as Jasmine reporter inside init.js:
-------------------------------------------------------------
jasmine.getEnv().addReporter(adapter);"
`;

exports[`DetoxRuntimeError should format all fields to a single message: throw error - message with hint and debug info 1`] = `
"DetoxRuntimeError: Invalid test summary was passed to detox.beforeEach(testSummary)
Expected to get an object of type: { title: string; fullName: string; status: \\"running\\" | \\"passed\\" | \\"failed\\"; }

HINT: Maybe you are still using an old undocumented signature detox.beforeEach(string, string, string) in init.js ?
See the article for the guidance: https://github.com/wix/detox/blob/master/docs/APIRef.TestLifecycle.md

testSummary was: \\"test name\\""
`;

exports[`DetoxRuntimeError should format all fields to a single message: throw error - no args 1`] = `"DetoxRuntimeError: "`;

exports[`DetoxRuntimeError should format all fields to a single message: throw error - only message 1`] = `"DetoxRuntimeError: The video is not being recorded on device (emulator-5554) at path: /sdcard/712398.mp4"`;
